avatars/avatars.js

```diff
@@ -1366,878 +170,344 @@ class Avatar {
-            const t1 = (now / 1000) % selectedAnimations[0].duration;
-            const src1 = selectedAnimations[0].interpolants[k];
-            const v1 = src1.evaluate(t1);
-
-            const t2 = (now / 1000) % selectedAnimations[1].duration;
-            const src2 = selectedAnimations[1].interpolants[k];
-            const v2 = src2.evaluate(t2);
-
-            target.fromArray(v1);
-            if (selectedAnimations[0].direction !== selectedAnimations[1].direction) {
-              // from Avaer: literally blending animation quaternions (spherical linear interpolation, slerp)
-              const distanceFactor = 1 - distance2 / totalDistance;
-              target.slerp(localQuaternion.fromArray(v2), distanceFactor);
-            }
-          };
-          _getHorizontalBlend(selectedAnimations, localQuaternion2);
-          _getHorizontalBlend(selectedOtherAnimations, localQuaternion3);
-          dst.copy(localQuaternion2).slerp(localQuaternion3, crouchFactor);
-        }
-        // blend
-        if (this.flyState || (this.flyTime >= 0 && this.flyTime < 1000)) {
-          const t2 = this.flyTime / 1000;
-          const f = this.flyState ? Math.min(cubicBezier(t2), 1) : (1 - Math.min(cubicBezier(t2), 1));
-          const src2 = floatAnimation.interpolants[k];
-          const v2 = src2.evaluate(t2 % floatAnimation.duration);
-
-          dst.slerp(localQuaternion.fromArray(v2), f);
-        }
+
+  _findAnimation(name){
+    for (let i=0;i<animations.length;i++){
+      if (animations[i].name===name){
+        return i
       }
     }
+    return -1
   }
 
-  _applyPose() {
-    if (poseData && (poseIndex !== -1 || this.activePoses.length > 0)) {
-      for (const k in this.outputs) {
-        this.outputs[k].quaternion.set(0, 0, 0, 1);
-      }
-
-      this.outputs.leftUpperArm.quaternion.setFromAxisAngle(forwardVector, Math.PI * 0.25);
-      this.outputs.rightUpperArm.quaternion.setFromAxisAngle(forwardVector, -Math.PI * 0.25);
-
-      if (poseIndex !== -1) {
-        for (const bone of poseData[poseIndex].bones) {
-          if (bone.mappedName) {
-            this.outputs[bone.mappedName].quaternion.premultiply(bone.quaternion);
-          }
-        }
-      }
-
-      for (const activePose of this.activePoses) {
-        const {index: poseIndex, value} = activePose;
-        for (const bone of poseData[poseIndex].bones) {
-          if (bone.mappedName) {
-            localQuaternion.copy(this.outputs[bone.mappedName].quaternion);
-            localQuaternion2.copy(this.outputs[bone.mappedName].quaternion).premultiply(bone.quaternion);
-            this.outputs[bone.mappedName].quaternion.copy(localQuaternion).slerp(localQuaternion2, value);
-          }
-        }
-      }
+  setAnimation(name,weight){
+    if (this.active_animations>=this.animations.length){
+      return
     }
-    if (this.headTargetEnabled) {
-      this.outputs.neck.quaternion.premultiply(this.headTarget);
+    if (this.getAnimation(name)!==-1){
```

Comment:
Why is there a two step finding of the animation?

Reply:
additional optimization plans